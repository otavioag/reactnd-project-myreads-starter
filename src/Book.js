import React from 'react';
import BookMenu from './BookMenu';

const Book = (props) => (
  <li>
    <div className='book'>
      <div className='book-top'>
        <div
          className='book-cover'
          style={{
            width: 128,
            height: 193,
            backgroundImage: 'url('+(props.book.imageLinks ? props.book.imageLinks.thumbnail : '')+')'
          }}
        />
        <BookMenu book={props.book} shelf={props.book.shelf ? props.book.shelf : 'none'} handleShelfChange={props.handleShelfChange} />
      </div>
      <div className='book-title'>{props.book.title}</div>
      {props.book.authors ? props.book.authors.map(author => (
        <div key={author} className='book-authors'>{author}</div>
      )) : ''}
    </div>
  </li>
);

export default Book;