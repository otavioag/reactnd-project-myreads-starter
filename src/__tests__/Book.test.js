import React from 'react';
import { shallow } from 'enzyme';
import Book from '../components/book/Book';

describe('Book', () => {
  it('renders without crashing - with book data', () => {
    const wrapper = shallow(<Book
      book={{title: "React", authors: ["Nils Hartmann", "Oliver Zeigermann"], imageLinks: {thumbnail: "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api"}, shelf: 'read'}}
      handleShelfChange={jest.fn()} />
    );
    expect(wrapper).toHaveLength(1);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders without crashing - without book data', () => {
    const wrapper = shallow(<Book
      book={{}}
      handleShelfChange={jest.fn()} />
    );
    expect(wrapper).toHaveLength(1);
  });
});