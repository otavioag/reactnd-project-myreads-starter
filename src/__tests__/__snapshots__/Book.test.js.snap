// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Book renders without crashing - with book data 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Book
    book={
      Object {
        "authors": Array [
          "Nils Hartmann",
          "Oliver Zeigermann",
        ],
        "imageLinks": Object {
          "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
        },
        "shelf": "read",
        "title": "React",
      }
    }
    handleShelfChange={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="book"
      >
        <div
          className="book-top"
        >
          <div
            className="book-cover"
            style={
              Object {
                "backgroundImage": "url(http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                "height": 193,
                "width": 128,
              }
            }
          />
          <BookMenu
            book={
              Object {
                "authors": Array [
                  "Nils Hartmann",
                  "Oliver Zeigermann",
                ],
                "imageLinks": Object {
                  "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                },
                "shelf": "read",
                "title": "React",
              }
            }
            handleShelfChange={[MockFunction]}
            shelf="read"
          />
        </div>
        <div
          className="book-title"
        >
          React
        </div>
        <div
          className="book-authors"
        >
          Nils Hartmann
        </div>
        <div
          className="book-authors"
        >
          Oliver Zeigermann
        </div>
      </div>,
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="book-top"
          >
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url(http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                  "height": 193,
                  "width": 128,
                }
              }
            />
            <BookMenu
              book={
                Object {
                  "authors": Array [
                    "Nils Hartmann",
                    "Oliver Zeigermann",
                  ],
                  "imageLinks": Object {
                    "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                  },
                  "shelf": "read",
                  "title": "React",
                }
              }
              handleShelfChange={[MockFunction]}
              shelf="read"
            />
          </div>,
          <div
            className="book-title"
          >
            React
          </div>,
          Array [
            <div
              className="book-authors"
            >
              Nils Hartmann
            </div>,
            <div
              className="book-authors"
            >
              Oliver Zeigermann
            </div>,
          ],
        ],
        "className": "book",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="book-cover"
                style={
                  Object {
                    "backgroundImage": "url(http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                    "height": 193,
                    "width": 128,
                  }
                }
              />,
              <BookMenu
                book={
                  Object {
                    "authors": Array [
                      "Nils Hartmann",
                      "Oliver Zeigermann",
                    ],
                    "imageLinks": Object {
                      "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                    },
                    "shelf": "read",
                    "title": "React",
                  }
                }
                handleShelfChange={[MockFunction]}
                shelf="read"
              />,
            ],
            "className": "book-top",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "book-cover",
                "style": Object {
                  "backgroundImage": "url(http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                  "height": 193,
                  "width": 128,
                },
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "book": Object {
                  "authors": Array [
                    "Nils Hartmann",
                    "Oliver Zeigermann",
                  ],
                  "imageLinks": Object {
                    "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                  },
                  "shelf": "read",
                  "title": "React",
                },
                "handleShelfChange": [MockFunction],
                "shelf": "read",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "React",
            "className": "book-title",
          },
          "ref": null,
          "rendered": "React",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Nils Hartmann",
          "nodeType": "host",
          "props": Object {
            "children": "Nils Hartmann",
            "className": "book-authors",
          },
          "ref": null,
          "rendered": "Nils Hartmann",
          "type": "div",
        },
        Object {
          "instance": null,
          "key": "Oliver Zeigermann",
          "nodeType": "host",
          "props": Object {
            "children": "Oliver Zeigermann",
            "className": "book-authors",
          },
          "ref": null,
          "rendered": "Oliver Zeigermann",
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "li",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="book"
        >
          <div
            className="book-top"
          >
            <div
              className="book-cover"
              style={
                Object {
                  "backgroundImage": "url(http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                  "height": 193,
                  "width": 128,
                }
              }
            />
            <BookMenu
              book={
                Object {
                  "authors": Array [
                    "Nils Hartmann",
                    "Oliver Zeigermann",
                  ],
                  "imageLinks": Object {
                    "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                  },
                  "shelf": "read",
                  "title": "React",
                }
              }
              handleShelfChange={[MockFunction]}
              shelf="read"
            />
          </div>
          <div
            className="book-title"
          >
            React
          </div>
          <div
            className="book-authors"
          >
            Nils Hartmann
          </div>
          <div
            className="book-authors"
          >
            Oliver Zeigermann
          </div>
        </div>,
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="book-top"
            >
              <div
                className="book-cover"
                style={
                  Object {
                    "backgroundImage": "url(http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                    "height": 193,
                    "width": 128,
                  }
                }
              />
              <BookMenu
                book={
                  Object {
                    "authors": Array [
                      "Nils Hartmann",
                      "Oliver Zeigermann",
                    ],
                    "imageLinks": Object {
                      "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                    },
                    "shelf": "read",
                    "title": "React",
                  }
                }
                handleShelfChange={[MockFunction]}
                shelf="read"
              />
            </div>,
            <div
              className="book-title"
            >
              React
            </div>,
            Array [
              <div
                className="book-authors"
              >
                Nils Hartmann
              </div>,
              <div
                className="book-authors"
              >
                Oliver Zeigermann
              </div>,
            ],
          ],
          "className": "book",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                <div
                  className="book-cover"
                  style={
                    Object {
                      "backgroundImage": "url(http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                      "height": 193,
                      "width": 128,
                    }
                  }
                />,
                <BookMenu
                  book={
                    Object {
                      "authors": Array [
                        "Nils Hartmann",
                        "Oliver Zeigermann",
                      ],
                      "imageLinks": Object {
                        "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                      },
                      "shelf": "read",
                      "title": "React",
                    }
                  }
                  handleShelfChange={[MockFunction]}
                  shelf="read"
                />,
              ],
              "className": "book-top",
            },
            "ref": null,
            "rendered": Array [
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "className": "book-cover",
                  "style": Object {
                    "backgroundImage": "url(http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api)",
                    "height": 193,
                    "width": 128,
                  },
                },
                "ref": null,
                "rendered": null,
                "type": "div",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "function",
                "props": Object {
                  "book": Object {
                    "authors": Array [
                      "Nils Hartmann",
                      "Oliver Zeigermann",
                    ],
                    "imageLinks": Object {
                      "thumbnail": "http://books.google.com/books/content?id=IOejDAAAQBAJ&printsec=frontcover&img=1&zoom=1&source=gbs_api",
                    },
                    "shelf": "read",
                    "title": "React",
                  },
                  "handleShelfChange": [MockFunction],
                  "shelf": "read",
                },
                "ref": null,
                "rendered": null,
                "type": [Function],
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "React",
              "className": "book-title",
            },
            "ref": null,
            "rendered": "React",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "Nils Hartmann",
            "nodeType": "host",
            "props": Object {
              "children": "Nils Hartmann",
              "className": "book-authors",
            },
            "ref": null,
            "rendered": "Nils Hartmann",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": "Oliver Zeigermann",
            "nodeType": "host",
            "props": Object {
              "children": "Oliver Zeigermann",
              "className": "book-authors",
            },
            "ref": null,
            "rendered": "Oliver Zeigermann",
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "li",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
